### DISCUSSION TITLE
jtx Board works as desktop application in GNU/Linux

### DISCUSSION THREAD

**Discussion by mcastillof** [TechbeeAT/jtxBoard#168] [Show and tell]
Hello, I was looking for a desktop app that had the same features of jtx Board, and I ended up installing jtx Board directly (well, a bit) on my GNU/Linux laptop.

The process involves installing Waydroid, a container-based approach to boot a full Android system on a regular GNU/Linux.
From there, installing jtx Board and DAVx5 from f-droid app or web site.

![image](https://user-images.githubusercontent.com/6936113/212321500-0d5e30d7-8aa5-404f-b82a-d53ddf2aa2b3.png)

**Comment by patrickunterwegs** [comment]
Hey @mcastillof , cool, thanks for that update 😎 
Did you also check if the sync through DAVx5 works in this Waydroid-container?

Ping @rfc2822 @devvv4ever

**Reply by devvv4ever** [reply]
Thats a cool thing! Didn't follow the WayDroid project in the last years. Does it offer contacts and calendar storages of Android as well?

**Reply by mcastillof** [reply]
Yes, DAVx5 sync is working. Contacts show up, and calendar events too.

**Reply by mcastillof** [reply]
some screenshots:
![image](https://user-images.githubusercontent.com/6936113/212347816-24441b1d-5d8e-4615-a288-8d29892a7fd4.png)
![image](https://user-images.githubusercontent.com/6936113/212348061-26ab116e-8a67-4a24-be28-622ce72e2efb.png)
![image](https://user-images.githubusercontent.com/6936113/212348913-06e7ea40-9eea-4943-82f4-ed304e5575da.png)
![image](https://user-images.githubusercontent.com/6936113/212349333-2f49657c-cc64-4728-9eca-aeec03a168f7.png)

**Comment by mcastillof** [comment]
My setup is GNU/Linux Debian Bookworm, Waydroid, LineageOS, F-Droid, DAVx5 and jtx Board.
You need Wayland. Clipboard works fine between the host machine and Waydroid, but mouse highlighting to copy and third mouse button to paste isn't.

**Comment by rfc2822** [comment]
Wow that's really cool!

**Comment by patrickunterwegs** [comment]
Hey @mcastillof , I would like to mention your approach to get jtx Board running on Twitter/Mastodon. If you like I could also mention you, any of your projects or blog 😉 Otherwise I'll just make a short post. Just let me know and I'll manage!

**Comment by mcastillof** [comment]
Hello Patrick, you and the FLOSS community deserve the credit for the approach. If you want, just mention my username ;)

**Comment by rogercreagh** [comment]
Aw shucks. I was really excited when I returned here and read this thread as it sounds like exactly what I've been looking for without having to install Android Studio. So I went off to waydro.io and followed the install instructions for Ubuntu and got shedloads of errors.

I think the problem is that Waydroid requires Wayland not X11, although it doesn't say that anywhere I can find (perhaps the clue is in the name). Wayland became the default on Ubuntu 22.04 but it is so buggy and doesn't work with many applications I use daily causing endless crashes and errors so I had to revert to X11 windowing.

**Reply by patrickunterwegs** [reply]
In fact I also tried to make it run and failed with different kinds of errors. After some hours I put it aside to have a look another time. But I hope that they'll make it easier and less error prone soon!

**Reply by mcastillof** [reply]
I had written an answer and send it, but I deleted it to repost it as an answer here, but I didn't had it in the clipboard, so it is gone. Maybe you received a copy of the original answer in your mail, from Github. If it is, please copy it here. If not, then my short  answer is, I will provide details about the difficults I had to setup Waydroid, and a detailed steps to install Waydroid in a freshly installed GNU/Linux Debian.

**Reply by rogercreagh** [reply]
@mcastillof was this the answer that got lost?

> Yes, it could be a difficult task, in my case because of the lack of Wayland and port 53 already in use, but it is possible.
> 
> Waydroid needs Wayland, so if you have X11 you can install Weston. This way you will have Wayland nested in your X11 session.
> 
> I also went the other way around and installed Gnome + Wayland to test it, and it also worked, and was better integrated. Every android app could be detached as a single window being displayed in your screen, instead of an Android window with apps inside. With Weston, you had, a Weston window, with an Android window inside with apps inside.
> 
> I had no Internet access from Waydroid because I had port 53 already in use, check with command ss or netstat. More info here: https://github.com/waydroid/waydroid/issues/117#issuecomment-1380760713
> 
> To copy and paste from the clipboard of your host OS to Waydroid and backwards you also had to install an app in your OS. I can't remember the name of the app, and if it worked with Weston and native Wayland.
> 
> I will try to provide detailed steps for a freshly installed Debian system.
>

**Reply by rogercreagh** [reply]
Yes indeed, the Weston solution is too poorly integrated for regular use. I can see that it works, but I didn't get as far as untangling the issue of getting wifi access within waydroid within weston.
I might try switching wayland back on and see if any of the issues I had have been fixed - most importantly getting Eclipse to work reliably and also drag'n'drop in Edge (and other) browsers - both were still misbehaving a couple of months ago and blame-wars between wayland, ubuntu, and eclipse developers on various fora as to the reasons and how (and more importantly who) to fix it.
For me switching to Wayland by Ubuntu was a classic example of don't replace a working component with a shiny new one until it is fully tested.

**Reply by mcastillof** [reply]
yes, that was the answer, thanks. I will provide the steps to get Waydroid running in a Debian machine in a few deays.

**Reply by mcastillof** [reply]
Another option to explore could be installing jtxBoard in Android-x86 in a virtual machine.

**Comment by boxallar** [comment]
You could also try using KOrganizer on Linux to sync with your jtx board on your phone. I believe it uses a lot of the same protocols.

**Comment by edellaporta** [comment]
Hi, i would like to know what service do you use to sync with your mobile? I've achieve to sync my tasks using mailbox but I couldn't with notes and journal.
